src/api-binder/report.ts

```diff
@@ -141,15 +142,30 @@ export async function startReporting() {
 	);
 	const doReport = async () => {
 		if (!reportPending) {
-			throttledReport(reportConfigs);
+			return throttledReport(reportConfigs);
 		}
 	};
 
 	// If the state changes, report it
 	deviceState.on('change', doReport);
-	// But check once every max report frequency to ensure that changes in system
-	// info are picked up (CPU temp etc)
-	setInterval(doReport, constants.maxReportFrequency);
-	// Try to perform a report right away
-	return doReport();
+
+	async function recursivelyReport(delayBy: number) {
+		// Convenience function used for delaying recursive reports
+		const delayedLoop = async () => {
+			// Wait until we want to report again
+			await delay(delayBy);
+			// Try to report again
+			await recursivelyReport(delayBy);
+		};
+
+		try {
+			// Try to send current state
+			await doReport();
+		} finally {
+			// Wait to report again
+			await delayedLoop();
+		}
```

Comment:
I think the extra function just makes things more confusing here because it's only used once in precisely one place and is so short that it's not really abstracting much useful (the amount of extra comments just to try to explain what this separate function is just adds more noise than in-lining it)
```suggestion
		try {
			// Try to send current state
			await doReport();
		} finally {
			// Wait until we want to report again
			await delay(delayBy);
			// Try to report again
			await recursivelyReport(delayBy);
		}
```